tv_apps/smart-system/fxa/js/screens/fxam_enter_email.js

```diff
@@ -196,6 +198,20 @@ var FxaModuleEnterEmail = (function() {
       }.bind(this)
     );
 
+    // There are 3 reasons why using setTimeout at this place:
+    // 1. Focus() only works in the setTimeout callback here
+    // 2. The input will be focused first and the keyboard will be brought
+    //    up. We need to do this after the slide up animation of the parent
+    //    fxa_dialog. But the fxa iframe has no way to know when the slide up
+    //    animation is finished.
+    // 3. Put the FxaModuleKeyNavigation.add in the onanimate callback in
```

Comment:
Yes I agreed hard code this is actually far from ideal. The problem we face is that the fxa_dialog slide in animation happens in the parent window outside, if we want to know exactly when the animation ends we will need to make other kinds of fix or workaround to pass the animation ended event to this child window.

So to add a fake animation that aligns to the animation duration of the parent window, we still need to hard code this duration in the fake animation unless we can receive the animation end event from the parent window.

The timeout value added here is mainly for smooth UI. Without it the functionality still works, a user will only see an animation glitch when fxa_dialog and the keyboard is brought up at the same time.